charts/publisher/templates/web/dbmigration-job.yaml

```diff
@@ -1,21 +1,23 @@
-{{- if .Values.dbMigrationEnabled -}}
+{{- if .Values.web.dbMigrationEnabled -}}
 apiVersion: batch/v1
 kind: Job
 metadata:
   name: {{ .Release.Name }}-dbmigrate-{{ now | unixEpoch }}
+  labels:
+    {{- include "publisher.labels" . | nindent 4 }}
+    app.kubernetes.io/component: "web"
 spec:
   template:
     metadata:
-      name: "{{ .Release.Name }}"
+      name: {{ .Release.Name }}-dbmigrate-{{ now | unixEpoch }}
       labels:
-        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
-        app.kubernetes.io/instance: {{ .Release.Name | quote }}
-        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        {{- include "publisher.labels" . | nindent 8 }}
+        app.kubernetes.io/component: "web"
     spec:
       restartPolicy: Never
       containers:
       - name: {{ .Release.Name }}-dbmigrate
-        image: "{{ .Values.appImage.repository }}:{{ .Values.appImage.tag }}"
+        image: "{{ .Values.web.image.repository }}:{{ .Values.web.image.tag }}"
```

Comment:
I'd recommend not allowing the image to differ between the "web" and "worker" deployments. By design they should always be running the same binary so it'd be best to model them that way in the chart, rather than potentially implying that the user could override them to be different from each other by passing in different values.

(In other words, if someone were to make use of these separate values by setting them to point at different images, that would always be a mistake - so it would be preferable to prevent the mistake by referring directly to common.image etc. in the Deployment specs. The idea being that it's preferable to make erroneous configurations impossible by design.)

Reply:
Fair enough - this was based on the principle that a worker process might not use exactly the same container as a web process, as the worker needs libraries that the web process doesn't (Sidekiq, resque, etc). That's obviously not the case just now though, so in the spirit of YAGNI I'll remove it - if that changes in the future the chart can obviously be updated.

Reply:
Cool, thanks for fixing. With Rails I doubt we'd want a separate worker image because all the tests implicitly assume that the workers and serving processes are running the same binary; the extra complexity and possibility of version drift would likely outweigh the benefit of a slightly smaller image for workers (which itself might not even be a benefit anyway, because having two images would reduce cache efficacy).